dbms/src/TiDB/Schema/SchemaGetter.cpp

```diff
@@ -172,7 +172,13 @@ void SchemaDiff::deserialize(const String & data)
         old_table_id = obj->getValue<Int64>("old_table_id");
         old_schema_id = obj->getValue<Int64>("old_schema_id");
 
-        regenerate_schema_map = obj->getValue<bool>("regenerate_schema_map");
+        if (obj->has("regenerate_schema_map"))
+        {
+            regenerate_schema_map = obj->getValue<bool>("regenerate_schema_map");
+        }
+
+        // Note If you want to add new fields here, please firstly check if it is present.
+        // Otherwise it will break when doing upgrading test.
```

Comment:
We'd better add a ut for parsing JSON string without "regenerate_schema_map". And it can avoid others adding fields without checking field existence.